cloud/services/roleassignments/roleassignments_test.go

```diff
@@ -0,0 +1,140 @@
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.SubscriptionID().AnyTimes().Return("12345")
+				s.ResourceGroup().Return("my-rg")
+				s.RoleAssignmentSpecs().Return([]azure.RoleAssignmentSpec{
+					{
+						MachineName: "test-vm",
+					},
+				})
+				v.Get(context.TODO(), "my-rg", "test-vm").Return(compute.VirtualMachine{
+					Identity: &compute.VirtualMachineIdentity{
+						PrincipalID: to.StringPtr("000"),
+					},
+				}, nil)
+				m.Create(context.TODO(), "/subscriptions/12345/", gomock.AssignableToTypeOf("uuid"), gomock.AssignableToTypeOf(authorization.RoleAssignmentCreateParameters{
+					Properties: &authorization.RoleAssignmentProperties{
+						RoleDefinitionID: to.StringPtr("/subscriptions/12345/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c"),
+						PrincipalID:      to.StringPtr("000"),
+					},
+				}))
+			},
+		},
+		{
+			name:          "error getting VM",
+			expectedError: "cannot get VM to assign role to system assigned identity: #: Internal Server Error: StatusCode=500",
+			expect: func(s *mock_roleassignments.MockRoleAssignmentScopeMockRecorder, m *mock_roleassignments.MockClientMockRecorder, v *mock_virtualmachines.MockClientMockRecorder) {
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.SubscriptionID().AnyTimes().Return("12345")
+				s.ResourceGroup().Return("my-rg")
+				s.RoleAssignmentSpecs().Return([]azure.RoleAssignmentSpec{
+					{
+						MachineName: "test-vm",
+					},
+				})
+				v.Get(context.TODO(), "my-rg", "test-vm").Return(compute.VirtualMachine{}, autorest.NewErrorWithResponse("", "", &http.Response{StatusCode: 500}, "Internal Server Error"))
+			},
+		},
+		{
+			name:          "return error when creating a role assignment",
+			expectedError: "cannot assign role to VM system assigned identity: #: Internal Server Error: StatusCode=500",
+			expect: func(s *mock_roleassignments.MockRoleAssignmentScopeMockRecorder, m *mock_roleassignments.MockClientMockRecorder, v *mock_virtualmachines.MockClientMockRecorder) {
+				s.V(gomock.AssignableToTypeOf(2)).AnyTimes().Return(klogr.New())
+				s.SubscriptionID().AnyTimes().Return("12345")
+				s.ResourceGroup().Return("my-rg")
+				s.RoleAssignmentSpecs().Return([]azure.RoleAssignmentSpec{
+					{
+						MachineName: "test-vm",
+					},
+				})
+				v.Get(context.TODO(), "my-rg", "test-vm").Return(compute.VirtualMachine{
+					Identity: &compute.VirtualMachineIdentity{
+						PrincipalID: to.StringPtr("000"),
+					},
+				}, nil)
+				m.Create(context.TODO(), "/subscriptions/12345/", gomock.AssignableToTypeOf("uuid"), gomock.AssignableToTypeOf(authorization.RoleAssignmentCreateParameters{})).Return(authorization.RoleAssignment{}, autorest.NewErrorWithResponse("", "", &http.Response{StatusCode: 500}, "Internal Server Error"))
+			},
+		},
+	}
+
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			tc := tc
```

Comment:
doesn't this need to be before the t.Run?

Reply:
yep it does